### PR TITLE
Update todo

### PR DESCRIPTION
**Pull request by hedita**

### CODE REVIEW

File: src/components/TodoItem/TodoItem.jsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
-import React from "react";
+import React, { useEffect } from "react";
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.jsx]: Import `useState` here.

File: src/components/TodoItem/TodoItem.jsx [hunk 2]
```diff
@@ -20,11 +23,67 @@ const TodoItem = ({ text, createdAt, taskId, getTasks }) => {
     }
   };
 
+  const updateStatus = () => {
+    try {
+      fetch(`${apiBaseUrl}/todos/${taskId}`, {
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.jsx]: fetch is a promise. await it.

File: src/components/TodoItem/TodoItem.jsx [hunk 3]
```diff
@@ -20,11 +23,67 @@ const TodoItem = ({ text, createdAt, taskId, getTasks }) => {
...
+        method: "PATCH",
+        body: JSON.stringify({ isDone }),
+        headers: requestDefaultHeaders,
+      });
+      setIsDone(!isDone);
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.jsx]: set the response of fetch here.

File: src/components/TodoItem/TodoItem.jsx [hunk 4]
```diff
@@ -20,11 +23,67 @@ const TodoItem = ({ text, createdAt, taskId, getTasks }) => {
...
+    } catch (error) {
+      setError(error.message);
+    }
+  };
+
+  const saveNewTodo = () => {
+    if (newText === text) {
+      setIsEditing(false);
+      return;
+    }
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.jsx:41]: What is the `return` use case here? Explain in a voice.

File: src/components/TodoItem/TodoItem.jsx [hunk 5]
```diff
@@ -20,11 +23,67 @@ const TodoItem = ({ text, createdAt, taskId, getTasks }) => {
...
+
+    try {
+      fetch(`${apiBaseUrl}/todos/${taskId}`, {
```
Comment (blacksrc) [inline src/components/TodoItem/TodoItem.jsx]: promise.

File: src/components/TodoItem/TodoItem.jsx (hunk 1 above)
Comment (hedita) [reply src/components/TodoItem/TodoItem.jsx]: Done.

File: src/components/TodoItem/TodoItem.jsx (hunk 2 above)
Comment (hedita) [reply src/components/TodoItem/TodoItem.jsx]: Done.

File: src/components/TodoItem/TodoItem.jsx (hunk 5 above)
Comment (hedita) [reply src/components/TodoItem/TodoItem.jsx]: Done.

File: src/components/TodoItem/TodoItem.jsx (hunk 4 above)
Comment (hedita) [reply src/components/TodoItem/TodoItem.jsx:41]: I sent a voice.

**Review by blacksrc — APPROVED** [review]

**Pull request merged**